Etabs_Adapter/CRUD/Read/Opening.cs

```diff
@@ -0,0 +1,130 @@
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using BH.Engine.Adapter;
+using BH.oM.Adapters.ETABS;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.SurfaceProperties;
+using BH.Engine.Adapters.ETABS;
+using BH.oM.Geometry;
+using BH.Engine.Geometry;
+using BH.oM.Adapters.ETABS.Elements;
+using BH.Engine.Structure;
+using BH.Engine.Spatial;
+
+
+namespace BH.Adapter.ETABS
+{
+#if Debug16 || Release16
+    public partial class ETABS2016Adapter : BHoMAdapter
+#elif Debug17 || Release17
+   public partial class ETABS17Adapter : BHoMAdapter
+#else
+    public partial class ETABSAdapter : BHoMAdapter
+#endif
+    {
+        /***************************************************/
+        /***    Read Methods                             ***/
+        /***************************************************/
+
+        private List<Opening> ReadOpening(List<string> ids = null)
+        {
+
+            List<string> openingNames = new List<string>();
+            List<Opening> openingList = new List<Opening>();
+
+            int nameCount = 0;
+            string[] nameArr = { };
+            m_model.AreaObj.GetNameList(ref nameCount, ref nameArr);
+
+
+            bool isOpening = false;
+            openingNames = nameArr.Where(panelName => {
+                m_model.AreaObj.GetOpening(panelName, ref isOpening);
+                return isOpening;
+            }).ToList();
+
+            ids = FilterIds(ids, openingNames);
+
+            foreach (string id in ids)
+            {
+                ETABSId etabsId = new ETABSId();
+                etabsId.Id = id;
+
+                Opening opening = new Opening();
+                Polyline pl = GetOpeningPerimeter(id);
+
+                opening.Edges = pl.SubParts().Select(x => new Edge { Curve = x }).ToList();
+
+                //Label and story
+                string label = "";
+                string story = "";
+                string guid = null;
+                if (m_model.AreaObj.GetLabelFromName(id, ref label, ref story) == 0)
+                {
+                    etabsId.Label = label;
+                    etabsId.Story = story;
+                }
+
+                if (m_model.AreaObj.GetGUID(id, ref guid) == 0)
+                    etabsId.PersistentId = guid;
+
+                opening.SetAdapterId(etabsId);
+                openingList.Add(opening);
+            }
+
+            return openingList;
+        }
+
+        /***************************************************/
+
+        private Polyline GetOpeningPerimeter(string id)
```

Comment:
```suggestion
        private Polyline GetOpeningOutline(string id)
```
I would change this so it's a little clearer, I was expecting it to return the perimeter value and wondered why you would need that 😂